cubix-sui-move/src/Cubix/Language/SuiMove/ParsePretty.hs

```diff
@@ -0,0 +1,167 @@
+module Cubix.Language.SuiMove.ParsePretty where
+
+import Control.Applicative (Alternative (..), optional)
+import Control.Monad (MonadPlus (..), forM_, unless, when)
+import Control.Monad.IO.Class (MonadIO (..))
+import Control.Monad.Reader (MonadReader, ReaderT (..), asks)
+import Control.Monad.State.Strict (MonadState (..), StateT (..), gets, modify')
+import Control.Monad.Trans.Maybe (MaybeT (..))
+import Data.ByteString (ByteString)
+import Data.ByteString qualified as BS
+import Data.ByteString.Char8 qualified as BSC
+import Data.IntMap.Strict (IntMap)
+import Data.IntMap.Strict qualified as IM
+import Data.IORef qualified as IORef
+import Data.Kind (Type)
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Maybe (isJust)
+import Data.Type.Equality (type (:~:) (..), type (:~~:) (..))
+
+import TreeSitter (NodeId (..), Range (..))
+import TreeSitter qualified as TS
+import TreeSitter.SuiMove (tree_sitter_sui_move)
+
+import Cubix.ParsePretty
+import Cubix.Language.SuiMove.Modularized
+
+parse :: FilePath -> IO (Maybe (MoveTerm SourceFileL))
+parse path = do
```

Comment:
There's a lot of steps in here. How much of it has to do with parsing Sui specifically? Separating out the TreeSitter-general stuff from the SuiMove-specific stuff would help a lot.